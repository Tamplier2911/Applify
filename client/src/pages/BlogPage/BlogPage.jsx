// import "./BlogPage.scss";
import React, { useEffect } from "react";
import { Switch, Route } from "react-router-dom";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  selectCurrentSet,
  selectAllSets,
  selectCurrentSlot,
  selectAllSlots,
  selectIsLoading,
} from "../../redux/blogs/blogs.selectors";
import { loadAllBlogpostsStart } from "../../redux/blogs/blogs.actions";

// components
import BlogsCollection from "../../components/BlogsCollection/BlogsCollection";
import BlogPost from "../../components/BlogPost/BlogPost";
import WithSpinnerHOC from "../../components/WithSpinnerHOC/WithSpinnerHOC";

// JS Rendering CSS
import { BlogPageContainer, BlogPagePlaceholder } from "./BlogPageStyles.js";

// buff blogs collection with spinner
const BlogsCollectionWithSpinner = WithSpinnerHOC(BlogsCollection);

const BlogPage = ({
  match,
  loadAllBlogpostsStart,
  isLoading,
  currentDataSet,
  allDataSets,
  currentDataSlot,
  allDataSlots,
}) => {
  useEffect(() => {
    if (!currentDataSet.length) {
      loadAllBlogpostsStart();
    }
  }, [loadAllBlogpostsStart, currentDataSet.length]);

  const blogsData = {
    currentDataSet,
    allDataSets,
    currentDataSlot,
    allDataSlots,
  };

  return (
    <BlogPageContainer data-test="blog-page">
      <Switch>
        <Route
          exact
          path={`${match.path}`}
          render={() => (
            <BlogsCollectionWithSpinner
              isLoading={isLoading}
              blogsData={blogsData}
            />
          )}
          data-test="blog-page-route-cl"
        />
        <Route
          path={`${match.path}/:id`}
          component={BlogPost}
          data-test="blog-page-route-id"
        />
      </Switch>
      <BlogPagePlaceholder data-test="blog-page-placeholder" />
    </BlogPageContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  currentDataSet: selectCurrentSet,
  allDataSets: selectAllSets,
  currentDataSlot: selectCurrentSlot,
  allDataSlots: selectAllSlots,
  isLoading: selectIsLoading,
});

export default connect(mapStateToProps, { loadAllBlogpostsStart })(BlogPage);
