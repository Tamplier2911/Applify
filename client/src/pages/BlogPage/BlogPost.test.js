import React from "react";

// component
import BlogPage from "./BlogPage";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<BlogPage />", () => {
  const initialStore = {
    blogs: {
      allBlogposts: [],
      allSets: {},
      currentSet: [],
      currentSlot: 0,
      allSlots: 0,
      isLoading: false,
      errorMessage: null,
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<BlogPage store={store} match={{ path: "/" }} />);
  });

  it("renders without error", () => {
    const blogPage = findByTestAttr(wrapper, "blog-page");
    expect(blogPage.length).toBe(1);
  });

  it("renders children blog-page-route-cl", () => {
    const blogPageRouteCl = findByTestAttr(wrapper, "blog-page-route-cl");
    expect(blogPageRouteCl.length).toBe(1);
  });

  it("renders children blog-page-route-id", () => {
    const blogPageRouteId = findByTestAttr(wrapper, "blog-page-route-id");
    expect(blogPageRouteId.length).toBe(1);
  });

  it("renders children blog-page-placeholder", () => {
    const blogPagePlaceholder = findByTestAttr(
      wrapper,
      "blog-page-placeholder"
    );
    expect(blogPagePlaceholder.length).toBe(1);
  });

  it("matches snapshot", () => {
    const blogPage = findByTestAttr(wrapper, "blog-page");
    expect(blogPage).toMatchSnapshot();
  });
});
