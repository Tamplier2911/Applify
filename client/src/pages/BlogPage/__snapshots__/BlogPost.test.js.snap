// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPage /> matches snapshot 1`] = `ShallowWrapper {}`;
